### ISSUE TITLE
To do ideas 2: define psi with a realistic H

### ISSUE THREAD

**Issue by jkingslake** [ldeo-glaciology/nye_eqns_bvp#2]
Define a realistic ice thickness H and use it to define the basic hydraulic gradient psi.

You can use the perfectly plastic model to define H as a simple first step - equate the driving stress to 100kpa and use the equation to define H as a parabola.

**Comment by glugeorge** [comment]
Just a progress update/sanity check. I am a bit suspect of my derivation of psi as it results in psi never being negative and blowing up as the glacier height reaches 0. 
I'm including screenshots of my derivation from my latex file (markup seems to have a hard time getting equations right, link to overleaf: https://www.overleaf.com/read/hzvvwvjkfwfj):
![image](https://user-images.githubusercontent.com/25989736/143138285-c8bd7443-b79a-461a-a4e9-e73c73203c71.png)
![image](https://user-images.githubusercontent.com/25989736/143138405-978d5640-66e8-46c8-96bf-859cc9e1bba6.png)

I'm using tau_d = 100kpa here. 
A quick comparison between the earlier psi (orange) and this one (blue) (all nondimensionalized) shows a pretty drastic difference. It also results in the channel closing at the lake terminus right away if I keep other parameters the same. Trying to sort through it, would love to hear your thoughts when you return!

**Comment by glugeorge** [comment]
A followup just to note down a shower thought - right now the profile is the way it is (assuming my math is right) because the glacier is the thickest at the ice dam and thinnest at the other end - perhaps I should define it in a different way such that its thickest point is a bit further away from the ice dam - would such a definition make sense?

**Comment by jkingslake** [comment]
I love seeing the equations written out like this. It makes it very clear, thanks!


>A quick comparison between the earlier psi (orange) and this one (blue)

I dont see the plots you refer to here.

But I can imagine that given our your initial difficulty finding a psi and _M_ that gave limit cycles, that introducing this drastically different psi will cause some very different behavour. 

One question is what you are doing with the ice dam thickness in the lake equation. Do you use H0 derived above?

> perhaps I should define it in a different way such that its thickest point is a bit further away from the ice dam - would such a definition make sense?

This would be an appropriate way of getting a psi which is negative at the lake (the same as using _b_ >0 in the parameterized psi you used previously). A couple of thoughts: 1. this is probably actually quite realistic, given that in real systems, ice is lost into the lake via calving and melting at the ice dam, so flow is partly diverted into the lake, 2. This might be difficult to simulate with a simple model, which means you would need to paramaterize it, and then you would be back to something similar to the previous approach and would lose some of the progress towards a more full ice-water coupling that you made by including a more realistic H. 

Fo now I would try reducing H0 to 100m (by changing tau_b) and changing _M_ to try get limit cycles.

**Comment by glugeorge** [comment]
Oh I forgot to include the quick plot of different psi values. Here it is:
![image](https://user-images.githubusercontent.com/25989736/143890144-532304c9-eff0-4954-8623-7a2d797a6188.png)
(xaxis is distance along channel, y axis is psi, both are nondimensionalized) 
I will try your suggestion of reducing H0 to 100m - I had used the H0 derived above. Using 100m would then also impose a new channel length that's a lot shorter than 10km since the glacier thickness would then drop to 0 a lot earlier along the channel.

**Comment by jkingslake** [comment]
the blow up in the new psi at the terminus is because the plastic approximation means that as H goes to 0 dH/dx goes to inf (as can be seen by eqn 8, when tau_b is uniform). This is unrealistic, particularly given that in reality, in a land-terminating glacier (which is the scenario we are considering here with H(s=10km) = 0), flow in the channel become open channel flow some distance upstream from the terminus. 

A simple correction would be to define a non-zero H at the terminus. 

>Using 100m would then also impose a new channel length that's a lot shorter...

yes, but only if you keep the same tau_b. You could also just change tau_b and keep channel length the same.

**Comment by glugeorge** [comment]
In my latest commit (dc16ddda9e4ff7eaddca4248193c98410c32c697), I got cycles again using psi defined by glacier height. However it took a lot of manipulating/restrictions. Here's a summary of changes screenshotted from the latex file:
![image](https://user-images.githubusercontent.com/25989736/144498923-fb6ceb95-901b-4e78-af4e-3b0966c33592.png)
Here are the resulting plots (screenshotted from the jupyter notebook stored on github):
![image](https://user-images.githubusercontent.com/25989736/144499174-4757b1ee-759b-4462-9704-2acf639b5a13.png)
![image](https://user-images.githubusercontent.com/25989736/144499302-ceb7fde6-538a-4768-8499-616266cab164.png)
For the velocity colormap, I omitted the boundary at channel terminus since that pre-set 1000m/y dominates the scale when compared to the other velocities.
